src/devices/video/mc6847.cpp

```diff
@@ -752,11 +793,36 @@ void mc6847_base_device::record_partial_body_scanline(uint16_t physical_scanline
 
 //-------------------------------------------------
 //  scanline_position_from_clock
+//
+//  The hsync ON interrupt marks the end of the
+//  previous line and the beginning of the next
+//  line.  Measuring clocks since hsync ON gives
+//  an offset into the current line.  This converts
+//  the clocks offset into a "position" offset for
+//  use by the record functions as index into
+//  the video_scanline arrays
 //-------------------------------------------------
 
 int32_t mc6847_base_device::scanline_position_from_clock(int32_t clocks_since_hsync)
 {
-	return (clocks_since_hsync - 20) / 4;
+	// This value determined by experimentation with the Dragon Fire game,
+	// which uses precise timing measurements to change video modes
+	// multiple times mid-line.  This value seems to minimize video
+	// garbage, but doesn't eliminate it, so there are likely timing
+	// issues elsewhere.
+	//
```

Comment:
Dragon Fire is under `drgnfire` in `coco_cart.xml, please use the romset ID here. I otherwise give it a shot later, I presume it's main system waitstates?

Reply:
Thanks for the info about the ROM.  I'd been using the disk version from here: https://colorcomputerarchive.com/repo/Disks/Games/Dragonfire%20(Tandy).zip (LOADM + EXEC).  I will look for the ROM and try it out next.  I haven't tried disassembling Dragonfire to see what it's doing, but debugging MAME itself (C++ debugger), I can see that Dragonfire uses the hsync ON interrupt.  My guess is that it responds to that interrupt by clearing a flag somewhere so that the mainline code is released at a known time to do its meticulously-timed processing for the next horizontal line.  Just a guess, though.

Reply:
Tried the cartridge version from https://colorcomputerarchive.com/repo/Cartridges/Dragon%20Fire%20(1984)%20(26-3098)%20(Tandy).ccc.  Performs exactly the same as the disk version before my change (moving & shimmering garbage), and the same as the disk version after my change (nearly no garbage).  Will add "drgnfire" to the comment.

Reply:
I've added "[Dragon Fire listed in coco_cart.xml as drgnfire.]" to the end of the comment.

Reply:
That's even more redundant now, just referring it as `coco_cart:drgnfire` or `drgnfire (coco_cart)` is enough. 
It's a QA convention, adding Long Name Game Or Software Title as a comment is not pretty to look at, plus it will eventually become outdated the moment somebody takes this (and PC-6001) to a serious testing level.

Reply:
I see.  Ok, I've replaced "dragon fire" with coco_cart:drgnfire.  There was also a prior comment at the top of the file that had referred to "dragon fire", so I replaced that as well.